src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineWindowsLauncher.java

```diff
@@ -113,15 +110,17 @@ protected void launch(ComputeEngineComputer computer, TaskListener listener, Ins
 
             SCPClient scp = conn.createSCPClient();
 
-            logInfo(computer, listener, "Copying agent.jar to: " + TMPDIR);
-            scp.put(Jenkins.get().getJnlpJars("agent.jar").readFully(), "agent.jar", TMPDIR);
+            String jenkinsDir = node.getRemoteFS();
+            logInfo(computer, listener, "Copying agent.jar to: " + jenkinsDir);
+            scp.put(Jenkins.get().getJnlpJars("agent.jar").readFully(), "agent.jar", jenkinsDir);
 
             // Confirm Java is installed
             if (!testCommand(computer, conn, "java -fullversion", logger, listener)) {
                 logWarning(computer, listener, "Java is not installed.");
                 return;
             }
 
+            String LAUNCHSTRING = "java -jar " + jenkinsDir + "\\agent.jar";
```

Comment:
Java style nit: camel case for `LAUNCHSTRING` since it is being moved to a local variable.

Reply:
ah good catch

Reply:
done.